daemon/api_apps.go

```diff
@@ -246,12 +335,20 @@ func postApps(c *Command, r *http.Request, user *auth.UserState) Response {
 		return InternalError("no services found")
 	}
 
+	// Now that we know the services we are affecting, do some additional checks/fixups
+	if err := validateScopeAgainstApps(inst, u, appInfos); err != nil {
```

Comment:
I'd tweak the name since this call is doing more than validation:

```suggestion
	if err := validateOrPatchScopeForApps(inst, u, appInfos); err != nil {
```